### PR TITLE
add ability to name floaterm terminals

### LINKED ISSUE #69: Ability to name floaterm sessions
I use asynctasks with floaterm. It will be useful if I can name some terms and send specific commands to specific terminals.

### PR DESCRIPTION
**Pull request by vn-ki**
```
FloatermNew name='test1'
FloatermToggle test1
FloatermSend test1
```

Backwards compatibility is maintained.

Closes #69

### CODE REVIEW

**Comment by vn-ki** [comment]
Should I add completions to toggle and send?

**Comment by voldikss** [comment]
> Should I add completions to toggle and send?

Thanks! It would be better to have completion for toggle and send, since we already have completion for `FloatermNew`

**Comment by vn-ki** [comment]
I also change the name of buffer local window options variable in this PR.

**Comment by vn-ki** [comment]
Also, the new arguments to the commands should be documented.

File: autoload/floaterm.vim [hunk 1]
```diff
@@ -109,10 +109,22 @@ function! floaterm#curr() abort
...
 
-function! floaterm#toggle()  abort
+function! floaterm#toggle(...)  abort
   if &filetype ==# 'floaterm'
     hide
   else
+    let termname = get(a:, 1, '')
+    if termname != ''
+      let bufnr = floaterm#terminal#get_bufnr(termname)
+      echo bufnr
```
[the remark this answers is not in the document]
Comment (vn-ki) [reply autoload/floaterm.vim:119]: I'm sorry :P

[the remark this answers is not in the document]
Comment (vn-ki) [reply autoload/floaterm.vim]: Fixed

**Comment by voldikss** [comment]
Is this ready to be merged?

**Comment by vn-ki** [comment]
Yes. If you don't want me to add anything more, this is done.

**Comment by voldikss** [comment]
Thank you for your contribution! It's really COOL!

**Pull request merged**